src/auditwheel/musllinux.py

```diff
@@ -17,21 +17,12 @@ class MuslVersion(NamedTuple):
 
 def find_musl_libc() -> pathlib.Path:
     try:
-        ldd = subprocess.check_output(["ldd", "/bin/ls"], errors="strict")
-    except (subprocess.CalledProcessError, FileNotFoundError):
-        LOG.error("Failed to determine libc version", exc_info=True)
+        (dl_path,) = list(pathlib.Path("/lib").glob("*ld-musl-*.so.1"))
```

Comment:
I might have missed a previous commit but why the leading `*` ?
If it's not needed, it's probably better to be stricter.
Shall the architecture part of the name be checked for consistency  ?
With what's in https://github.com/pypa/auditwheel/blob/353c24250d66951d5ac7e60b97471a6da76c123f/src/auditwheel/policy/__init__.py#L82-L91

Reply:
Shall be look for `ld` or  `libc` ? I'd rather look for libc as it allows to use the same look-up in policies & here.

Reply:
I've updated this to look for libc instead. On musl libc is a symlink to the dynamic loader, but this should be fine.

I'm not sure about the architecture check. I think to reuse the `soname_map` in both places, we'd have to move `find_musl_libc` into the policy module to avoid a circular import. It also requires refactoring `_fixup_musl_libc_soname`. I think this change should be done separately.

Reply:
let's do this separately then.